python/src/scipp/compat/mantid.py

```diff
@@ -878,13 +878,26 @@ def to_workspace_2d(x, y, e, coord_dim, instrument_file=None):
             "as detailed in the installation instructions (https://scipp."
             "github.io/getting-started/installation.html)")
 
-    assert len(y.shape) == 2, "Currently can only handle 2D data."
+    assert (len(y.shape) == 2 or len(y.shape) == 1), \
+        "Currently can only handle 2D data."
 
     e = np.sqrt(e) if e is not None else np.sqrt(y)
 
+    # Convert a single array (e.g. single spectra) into 2d format
+    if len(y.shape) == 1:
+        y = np.array([y])
+
+    if len(e.shape) == 1:
+        e = np.array([e])
+
     unitX = validate_dim_and_get_mantid_string(coord_dim)
 
     nspec = y.shape[0]
+    if len(x.shape) == 1:
+        # SCIPP is using a  1:n spectra coord mapping, Mantid needs
+        # a 1:1 mapping so expand this out
+        x = np.array([x] * nspec)
+
```

Comment:
Does it? Mantid has shared ("common") X coords. Is it the way we are creating the workspace not supporting creating such as workspace with shared X?

Reply:
So `setSharedX` exists in the CPP layer, but is not exposed through Python binding layer, so currently the best we can do is:

```
if len(x.shape) == 1:
    for i in range(nspec):
        ws.setX(i,x)
else:
    for i in range(nspec):
        ws.setX(i, x[i])
```

Ideally, if this causes problems we should expose it on the Mantid side then come back and update this, but that blocks this PR until work is done on Mantid

Reply:
Well, we should bother with changing Mantid here I think, but at least use `numpy.broadcast_to` so you don't need to allocate a lot of memory.

Reply:
Done thanks :+1: